python/lance/_lib.pyx

```diff
@@ -363,6 +384,34 @@ cdef class FileSystemDataset(Dataset):
         )
         return FileSystemDataset.wrap(static_pointer_cast[CDataset, CLanceDataset](dataset))
 
+    def diff(self, v1: int, v2: int) -> LanceDiff:
+        """
+        Get the difference from v1 to v2 of this dataset
+
+        Parameters
+        ----------
+        v1: int
+            Start version
+        v2: int
+            End version
+        """
+        return LanceDiff(self.checkout(v1), self.checkout(v2))
+
+    def compute_metric(self, metric_func: Callable[[Dataset], pd.DataFrame],
+                       versions: list[int] = None) -> pd.DataFrame:
+        """
+        Compute a metric across versions for this dataset
```

Comment:
Main concerns are the stability / maturity of these two APIs.

Reply:
as discussed, moved to `lance/__init__.py` and not as a Dataset method